File: src/main/java/TokenType.java
public enum TokenType {
    AND,
    OR,
    NOT;

    public static TokenType getTokenType(String token) {
        if (token.startsWith("+")){
          return TokenType.OR;
        }
        else if (token.startsWith("-")){
          return TokenType.NOT;
        }
        return TokenType.AND;
    }
}

File: Phase05/InvertedIndex/TokenType.cs
namespace InvertedIndex;

public enum TokenType
{
    And,
    Or,
    Not
}

public static class TokenMethods
{

    public static TokenType GetTokenType(this string token)
    {
        if (token.StartsWith("+"))
        {
            return TokenType.Or;
        } else if (token.StartsWith("-"))
        {
            return TokenType.Not;
        }

        return TokenType.And;
    }
}